src/wp-admin/site-editor.php

```diff
@@ -120,7 +120,7 @@ static function( $classes ) {
 
 wp_add_inline_script(
 	'wp-blocks',
-	sprintf( 'wp.blocks.setCategories( %s );', wp_json_encode( get_block_categories( $post ) ) ),
+	sprintf( 'wp.blocks.setCategories( %s );', wp_json_encode( get_block_categories( $block_editor_context ) ) ),
```

Comment:
https://github.com/WordPress/wordpress-develop/blob/c3ff3dbfc434bf53390f013102cf1ab86411ef0a/src/wp-includes/block-editor.php#L384

We already have block categories correctly computed in `$editor_setting` in the `get_block_editor_settings` call. Maybe we could reuse it instead here and in all other places where `wp.blocks.setCategories` needs to be set.

There is also this question if we really need to call `wp.blocks.setCategories` here. Maybe we could do it on the client using the passed `$editor_settings` object. This could be a larger refactor, though.

Reply:
Thanks for the suggestion, @gziolo. I've updated the logic.

> There is also this question if we really need to call wp.blocks.setCategories here. Maybe we could do it on the client using the passed $editor_settings object. This could be a larger refactor, though.

Right, this might be a larger refactor, and we should probably do it in Gutenberg first.